indico/web/client/js/react/components/PublicationSwitch.jsx

```diff
@@ -0,0 +1,79 @@
+import publicationURL from 'indico-url:contributions.manage_publication';
+
+import React, {useState, useEffect} from 'react';
+import PropTypes from 'prop-types';
+import {Header, Modal, Button, Checkbox} from 'semantic-ui-react';
+import indicoAxios, {handleAxiosError} from 'indico/utils/axios';
+
+export default function PublicationSwitch({confId}) {
+  const [published, setPublished] = useState(true);
+  const [modalOpen, setModalOpen] = useState(false);
+
+  useEffect(() => {
+    const fetchPublicationSetting = async () => {
```

Comment:
Is there any reason why this function has been defined inside the `useEffect` callback? Even though this effect will not be called more than once (since `confId` will not change) I think it would be nice to move the function outside of the callback and then you could also remove the `confId` from the `useEffect` dependencies.

Reply:
You could also move it outside the component and pass the `eventId` as an argument.

But I think the `eventId` *should* remain in the effect's dependency list, because in the (not-gonna-happen) case of the prop changing you DO need the effect to re-run and fetch the state of the new event.

Reply:
I think the `eventId` should stay in the dependency list. In the case of the prop changing to another ID (even if that's not gonna happen), you want to reload the data for the new ID.